### ISSUE TITLE
Existing flow runs duplicated when DST ends

### ISSUE THREAD

**Issue by drfraser** [PrefectHQ/prefect#7381]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

A couple of days ago, I regenerated deployments, reset the database (it was getting too big) and flow runs were created and scheduled up to and on the 30th or so - I did not examine things as it has always worked without any issues.

The deployments are cron based and the timezone is set to Europe/London as is the server.

Today, after the end of BST, it seems duplicate flow runs were automatically generated around midnight for today (the 30th) based on the flow run attributes - i.e. the already existing ones for the 30th were ignored and new ones created that ran an hour later. As these flows were designed to run once a day, the 2nd ones were crashing. No data loss, given how the flows work, but annoying to see failed runs in the UI...

This of course is unexpected, though it makes some sense. IMO, the generation of flow runs process should take into account whether time has shifted to DST ending or starting and manage already existing runs that might be "equivalents" accordingly. 

There were no problems with the one flow scheduled to run at an interval on every hour.  Hopefully this is just a problem with my version of Prefect and upgrading to the latest will fix this, but I could not find any existing issues...

 

### Reproduction

```python3
1. Create a flow + deployment and schedule the flow for every day at a specific time.
Do not use UTC as the timezone but one that uses DST - e.g. Europe/London

2. Wait until DST is over or starts, then see if Prefect has automatically created a duplicate flow run for the flow that is an hour later (or earlier)
```

### Error

```python3
Flow runs scheduled to run only once a day were automatically duplicated upon the end of DST and caused problems.
```

### Versions

```Text
Version:             2.0.4
API version:         0.8.0
Python version:      3.8.10
Git commit:          39db6fb1
Built:               Wed, Aug 10, 2022 1:19 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         ephemeral
Server:
  Database:          sqlite
  SQLite version:    3.31.1
```

**Comment by vsytchenko-exos** [comment]
Faced the same issue. Several pipelines were scheduled to run once a day with europe/warsaw timezone. 29-30th of  October was a switch to winter time. One hour back. And I've noticed this:
<img width="1368" alt="Screenshot 2022-10-31 at 01 45 34" src="https://user-images.githubusercontent.com/107683703/198910624-d7297a3e-3c10-402a-b8cc-160bbe8bece8.png">

**Comment by madkinsz** [comment]
@zangell44 / @jlowin is this a bug with the scheduler idempotency keys or schedule itself?

**Comment by MrJack91** [comment]
Please fix this.
This is critical...

We faced this issue every 6 months. This leads to massive problems.

**Comment by drfraser** [comment]
I updated my Prefect setup to the latest version before the timezone change and did not see this problem when DST came around. So it's fixed as far as I care, but I will check to see if what jlowin mentions applies to my setup.

**Comment by desertaxle** [comment]
Closing this since the scheduler's date generators have been reworked with explicit DST deduplication guards and DST-aware datetime arithmetic. The last commenter confirmed that updating to a newer version resolved the issue. If anyone sees duplicate scheduled runs across a DST boundary on a current version of Prefect, please open a new issue with details on the schedule configuration.
